src/lexer.rs

```diff
@@ -11,18 +11,27 @@ pub struct Lexer<'a> {
 	indentation: usize,
 	target_indent: usize,
 	new_line: bool,
+	peeked: Option<Token>,
 }
 
 impl<'a> Lexer<'a> {
 	pub fn new(string: &'a str) -> Self {
 		let characters = string.char_indices().peekable();
-		Lexer { string, characters, indentation: 0, target_indent: 0, new_line: true }
+		Lexer { string, characters, indentation: 0, target_indent: 0, new_line: true, peeked: None }
 	}
 
 	pub fn token(&mut self) -> Result<Option<Token>, ParserError> {
+		if self.peeked.is_some() {
```

Comment:
Is it possible to change this to a match statement like so:
```
match self.peeked.take() {
    Some(token) => token,
    None => self.next().transpose(),
}
```

Reply:
Done.